test/modules/generic-mqtt-tester/src/settings.rs

```diff
@@ -35,6 +38,11 @@ impl Settings {
         config.merge(File::from_str(DEFAULTS, FileFormat::Json))?;
         config.merge(Environment::new())?;
 
+        let test_scenario: TestScenario = config.get("test_scenario")?;
+        if let TestScenario::Initiate = test_scenario {
+            config.set("batch_id", Some(Uuid::new_v4().to_string()))?;
```

Comment:
Config doesn't support Uuid directly so we need to convert to string. Then we convert back to Uuid in the getter.

Reply:
I thought that Config supports anything that is deserializable. What is the error you're getting?

Reply:
On the set method when I remove to_string():
```
error[E0277]: the trait bound `config::value::ValueKind: From<Uuid>` is not satisfied
  --> src/settings.rs:43:20
   |
43 |             config.set("batch_id", Some(Uuid::new_v4()))?;
   |                    ^^^ the trait `From<Uuid>` is not implemented for `config::value::ValueKind`
   |
   = help: the following implementations were found:
             <config::value::ValueKind as From<&'a str>>
             <config::value::ValueKind as From<HashMap<std::string::String, T>>>
             <config::value::ValueKind as From<Vec<T>>>
             <config::value::ValueKind as From<bool>>
           and 4 others
   = note: required because of the requirements on the impl of `Into<config::value::ValueKind>` for `Uuid`
   = note: required because of the requirements on the impl of `From<std::option::Option<Uuid>>` for `config::value::ValueKind`
   = note: required because of the requirements on the impl of `Into<config::value::ValueKind>` for `std::option::Option<Uuid>`
   = note: required because of the requirements on the impl of `From<std::option::Option<Uuid>>` for `config::Value`
   = note: required because of the requirements on the impl of `Into<config::Value>` for `std::option::Option<Uuid>`
```